spec/requests/chat_channel_controller_spec.rb

```diff
@@ -291,4 +291,91 @@
       expect(membership.mobile_notification_level).to eq("never")
     end
   end
+
+  describe "#for_tag" do
+    fab!(:tag_with_channel) { Fabricate(:tag) }
+    fab!(:tag_without_channel) { Fabricate(:tag) }
+    fab!(:tag_channel) { Fabricate(:chat_channel, chatable: tag_with_channel) }
+
+    it "errors for anon" do
+      get "/chat/chat_channels/for_tag/#{tag_with_channel.name}.json"
+      expect(response.status).to eq(403)
+    end
+
+    it "returns the tag's chat channel" do
+      sign_in(user)
+      get "/chat/chat_channels/for_tag/#{tag_with_channel.name}.json"
+      expect(response.status).to eq(200)
+      expect(response.parsed_body["chat_channel"]["id"]).to eq(tag_channel.id)
+    end
+
+    it "returns nil if a chat channel doesn't exist for tag" do
+      sign_in(user)
+      get "/chat/chat_channels/for_tag/#{tag_without_channel.name}.json"
+      expect(response.status).to eq(200)
+      expect(response.parsed_body["chat_channel"]).to be_nil
+    end
+
+    describe "hidden tag" do
+      let(:hidden_tag_name) { "hidden1" }
+      before do
+        create_hidden_tags([hidden_tag_name])
+        ChatChannel.create(chatable: Tag.find_by(name: hidden_tag_name))
+      end
+
+      it "does not return the channel for normal user" do
+        sign_in(user)
+        get "/chat/chat_channels/for_tag/#{hidden_tag_name}.json"
+        expect(response.status).to eq(403)
+      end
+
+      it "returns the channel for admin" do
+        sign_in(admin)
+        get "/chat/chat_channels/for_tag/#{hidden_tag_name}.json"
+        expect(response.status).to eq(200)
+        expect(response.parsed_body["chat_channel"]).not_to be_nil
```

Comment:
I think it'll be better to assert for the right `chat_channel` instead of just `not_to be_nil`. The response can be returning else in the future and this test will still pass.